src/v/kafka/server/protocol.cc

```diff
@@ -60,7 +61,33 @@ protocol::protocol(
       config::shard_local_cfg().enable_idempotence.value())
   , _credentials(credentials)
   , _authorizer(authorizer)
-  , _security_frontend(sec_fe) {}
+  , _security_frontend(sec_fe) {
+    /*
+     * start the queue depth control algorithm. on each timer invocation we
+     * advance the latency tracker and optionally update the queue depth.
+     */
+    auto depth_update_freq
+      = config::shard_local_cfg().kafka_qdc_depth_update_ms();
```

Comment:
we'be been trying to get away from depending on this global state and using structs or args to capture them  in the ctor for a while.

Reply:
this isn't reusable.

Reply:
i think this answers a different comment? i don't get the answer.

Reply:
ah i red below.